sdk/core/System.ClientModel/src/Pipeline/ResponseBufferingPolicy.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// Pipeline policy to buffer response content or add a timeout to response content
+/// managed by the client.
+/// </summary>
+public class ResponseBufferingPolicy : PipelinePolicy
```

Comment:
Is the class public so that Azure.Core can use it?

Reply:
Yes.